petab/v2/problem.py

```diff
@@ -979,22 +973,12 @@ def add_parameter(
             record[LOWER_BOUND] = lb
         if ub is not None:
             record[UPPER_BOUND] = ub
-        if init_prior_type is not None:
-            record[INITIALIZATION_PRIOR_TYPE] = init_prior_type
-        if init_prior_pars is not None:
-            if not isinstance(init_prior_pars, str):
-                init_prior_pars = PARAMETER_SEPARATOR.join(
-                    map(str, init_prior_pars)
-                )
-            record[INITIALIZATION_PRIOR_PARAMETERS] = init_prior_pars
-        if obj_prior_type is not None:
-            record[OBJECTIVE_PRIOR_TYPE] = obj_prior_type
-        if obj_prior_pars is not None:
-            if not isinstance(obj_prior_pars, str):
-                obj_prior_pars = PARAMETER_SEPARATOR.join(
-                    map(str, obj_prior_pars)
-                )
-            record[OBJECTIVE_PRIOR_PARAMETERS] = obj_prior_pars
+        if prior_dist is not None:
+            record[PRIOR_DISTRIBUTION] = prior_dist
+        if prior_pars is not None:
+            if not isinstance(prior_pars, str):
+                prior_pars = PARAMETER_SEPARATOR.join(map(str, prior_pars))
+            record[PRIOR_PARAMETERS] = prior_pars
```

Comment:
```suggestion
        if isinstance(prior_pars, Sequence):
            record[PRIOR_PARAMETERS] = PARAMETER_SEPARATOR.join(map(str, prior_pars))
```

Reply:
I would have relied on the validation in `Parameter`.

Reply:
Oops, prior parameters where missing there altogether...